src/devices/lacrosse_tx22uit.c

```diff
@@ -0,0 +1,135 @@
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/**
+LaCrosse Technology View TX22U-IT temperature, humidity, wind speed/direction
+and rain sensor
+
+Product pages:
+TBW
+
+Specifications:
+TODO: update for TX22U-IT
+- Wind Speed Range: 0 to 178 kmh
+- Degrees of Direction: 360 deg with 16 Cardinal Directions
+- Outdoor Temperature Range: -29 C to 60 C
+- Outdoor Humidity Range: 10 to 99 %RH
+- Update Interval: Every 31 Seconds
+
+http://nikseresht.com/blog/?p=99 tell protocol but my TX22U-IT is different protocol.
+*/
+
+#include <stdbool.h>
+#include "decoder.h"
+
+static int lacrosse_tx22uit_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[] = { 0xaa, 0xaa, 0x2d, 0xd4 };
+
+    data_t *data;
+    uint8_t *b = bitbuffer->bb[0];
+    int offset;
+    static int raw_temp = 0, humidity = -1, raw_speed = -1, direction = -1;
+    float temp_c, speed_kmh;
+    bool upload_data = false;
+
+    offset = bitbuffer_search(bitbuffer, 0, 0,
+            preamble_pattern, sizeof(preamble_pattern) * 8);
+
+    if (offset >= bitbuffer->bits_per_row[0]) {
+        if (decoder->verbose) {
+            fprintf(stderr, "%s: Sync word not found\n", __func__);
+        }
+        return DECODE_ABORT_EARLY;
+    }
+
+#if 1 /* TODO: dump bitbuffer only on verbose */
+    bitbuffer_print(bitbuffer);
+#else
+    if (decoder->verbose) {
+        bitbuffer_print(bitbuffer);
+    }
+#endif
+    /*
+       0  1  2  3  4  5  6  7  8  9 10 11 12 13 14
+      aa aa 2d d4 ad 03 05 92 20 01 3d 00 b0 00 00  19.2C  wind 13 at 0.0km/h
+      aa aa 2d d4 ad 02 20 01 40 00 1f 00 00
+      aa aa 2d d4 ad 01 20 01 a3 00 00
+      aa aa 2d d4 ad 02 10 56 20 01 86 00 00        56%
+   */
+    for (offset = 6; offset < bitbuffer->bits_per_row[0] / 8; offset += 2) {
+        switch (b[offset] >> 4) {
+        case 0: /* temperature */
+            raw_temp = (b[offset] & 0x0f) * 100 + ((b[offset+1] >> 4) * 10) +
+              (b[offset+1] & 0x0f);
+            upload_data = true;
+            break;
+        case 1: /* humidity  */
+            humidity = (b[offset] & 0x0f) * 100 + (b[offset+1] >> 4) * 10 +
+              (b[offset+1] & 0x0f);
+            break;
+        case 3: /* wind */
+            direction = b[offset] & 0x0f;
+            raw_speed = ((b[offset+1] >> 4) * 10) + (b[offset+1] & 0x0f);
+            break;
+        }
+    }
+    if (upload_data) {
+        temp_c = (raw_temp - 400) * 0.1f;
+        speed_kmh = raw_speed * 0.1f;
+
+        if (humidity < 0 || humidity > 100
+            || temp_c < -40 || temp_c > 70
+            || direction < 0 || direction > 360
+            || speed_kmh < 0 || speed_kmh > 200)
+            return DECODE_FAIL_SANITY;
+
+        data = data_make(
```

Comment:
wrap data_make in `/* clang-format off */`

Reply:
will be wrapped.
Thx.